pkg/rootless/rootless_linux.c

```diff
@@ -311,10 +311,10 @@ static void __attribute__((constructor)) init()
         do_socket_activation = true;
         saved_systemd_listen_pid = strdup(listen_pid);
         saved_systemd_listen_fds = strdup(listen_fds);
-        saved_systemd_listen_fdnames = strdup(listen_fdnames);
+        if (listen_fdnames != NULL)
+          saved_systemd_listen_fdnames = strdup(listen_fdnames);
```

Comment:
```suggestion
        saved_systemd_listen_fdnames = NULL;
        if (listen_fdnames != NULL)
          saved_systemd_listen_fdnames = strdup(listen_fdnames);
```

Wouldn't it otherwise lead to undefined behavior when it's later used? I guess the initializing could be done in different ways (for instance at the same time as it is declared).



Another thing I thought about is that there is no `free()` of the `strdup()` strings.

Quote from `man strdup()`:
_Memory for the new string is obtained with malloc(3), and can be freed with free(3)._

Maybe that is not so important though.